apps/dashboard/src/components/billing/highlights-row.tsx

```diff
@@ -1,33 +1,50 @@
-import { Badge } from '@/components/primitives/badge';
-import { Card } from '@/components/primitives/card';
-import { ApiServiceLevelEnum } from '@novu/shared';
+import { Badge } from "@/components/primitives/badge";
+import { Card } from "@/components/primitives/card";
+import {
+  ApiServiceLevelEnum,
+  FeatureFlags,
+  FeatureFlagsKeysEnum,
+  FeatureNameEnum,
+  getFeatureForTierAsNumber,
+  getFeatureForTierAsText
+} from "@novu/shared";
+import { useFlagsMap } from "@/hooks/use-feature-flag.tsx";
 
-interface Highlight {
-  text: string;
-  badgeLabel?: string;
+const serviceLevelHighlightFunctions: Record<
+  string,
+  ((ApiServiceLevelEnum: ApiServiceLevelEnum, activeFlags: FeatureFlags) => string)[]
+> = {
+  [ApiServiceLevelEnum.FREE]: [getEventsLine, getTeammatesLine, feedRetentionLine],
+  [ApiServiceLevelEnum.PRO]: [getEventsLine, getTeammatesLine, feedRetentionLine],
+  [ApiServiceLevelEnum.TEAM]: [getEventsLine, getTeammatesLine, feedRetentionLine],
+  [ApiServiceLevelEnum.ENTERPRISE]: [getEventsLine, getTeammatesLine, getSamlText],
+};
+
+function augmentBasedOfFeatureFlags(highlightsArray: Partial<PlanHighlights>, featureFlags: FeatureFlags) {
+  if (!featureFlags[FeatureFlagsKeysEnum.IS_2025_Q1_TIERING_ENABLED]) {
```

Comment:
Consider using the `useFeatureFlag` to read the feature flad to avoid introducing a new pattern on feature flag consumption.

Reply:
Instead of three functions manipulating arrays of functions per plan, having two objects with the highlights per plan, one before and one after, the feature flag enablement is more readable and less prone to runtime bugs. 

That way, the future reader will have to comprehend a single condition based on a feature flag, and post-release, we will delete the legacy highlight object.

Reply:
Moreover, regarding the feature flags, since we have 1-2 feature flags controlling the new packaging release, we should simplify the logic instead of passing the while feature flag map to these functions.